warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BaseIndexExpansionVisitor.java

```diff
@@ -187,9 +190,37 @@ protected void rebuildFutureJexlNodes() {
             // if the parent is not null, replace the child
             // if the parent is null, this is the root node, and we will handle that in the expand method
             if (futureJexlNode.jjtGetParent() != null) {
-                JexlNodes.replaceChild(futureJexlNode.jjtGetParent(), futureJexlNode, newNode);
+                if (!(newNode instanceof ASTOrNode || newNode instanceof ASTAndNode)) {
```

Comment:
would love to see a bit more on why this is necessary and what you're correcting.

Reply:
removed in next commit